src/main/java/in/org/projecteka/hiu/consent/ConsentService.java

```diff
@@ -205,4 +229,68 @@
                         .switchIfEmpty(Mono.error(consentArtefactNotFound())))
                 .collectList();
     }
+
+    public Mono<Void> createRequest(String requesterId, ConsentRequestData consentRequestData) {
+        return validateConsentRequest(consentRequestData)
+                .then(sendConsentRequestToGateway(requesterId, consentRequestData));
+    }
+
+    private Mono<Void> sendConsentRequestToGateway(
+            String requesterId,
+            ConsentRequestData consentRequestData) {
+        String cmSuffix = getCmSuffix(consentRequestData.getConsent());
+        var requestDetail = consentRequestData.getConsent().to(
+                requesterId,
+                hiuProperties.getId(),
+                conceptValidator);
+        var gatewayRequestId = UUID.randomUUID();
+        return centralRegistry.token()
+                .flatMap(token -> {
+                    return gatewayServiceClient.sendConsentRequest(
+                            token, cmSuffix,
+                            ConsentRequest.builder()
+                                    .requestId(gatewayRequestId)
+                                    .timestamp(java.time.Instant.now().toString())
+                                    .consent(requestDetail)
+                                    .build());
+                }).then(consentRepository.insertConsentRequestToGateway(
+                        consentRequestData.getConsent().toConsentRequest(gatewayRequestId.toString(), requesterId)));
+    }
+
+    private String getCmSuffix(Consent consent) {
+        String[] parts = consent.getPatient().getId().split("@");
+        return parts[1];
+    }
+
+    public Mono<Void> updatePostedRequest(ConsentRequestInitResponse response) {
+        if (response.getError() != null) {
+            logger.error(String.format("[ConsentService] Received error response from consent-request. HIU RequestId=%s, Error code = %d, message=%s",
+                    response.getResp().getRequestId(),
+                    response.getError().getCode(),
+                    response.getError().getMessage()));
+            return consentRepository.updateConsentRequestStatus(
+                    response.getResp().getRequestId(),
+                    ConsentStatus.ERRORED,
+                    "");
+        }
+
+        if (response.getConsentRequest() != null) {
+            return consentRepository.consentRequestStatus(response.getResp().getRequestId())
+                    .switchIfEmpty(Mono.error(consentRequestNotFound()))
+                    .flatMap(status -> updateConsentRequestStatus(response, status));
+        } else {
```

Comment:
We don't actually need this `else` block here as we are returning from `if` block.

Reply:
done